File: frontend/src/app/core/services/admin-system-health.service.ts
import { Injectable, inject } from '@angular/core';
import { ApiService } from './api.service';
import {
  Subsystem,
  SystemHealthKpis,
  SystemHealthEvent,
  MaintenanceMode,
} from '../../models/admin-system-health.model';

/**
 * Platform-staff API client for System Health: live subsystem probes, KPI totals, recent-events feed,
 * per-subsystem detail/re-probe, cache flush, and the maintenance-mode toggle. Reads need INFRA_VIEW;
 * mutations need INFRA_MANAGE (enforced + audit-logged on the backend).
 */
@Injectable({ providedIn: 'root' })
export class AdminSystemHealthService {
  private readonly api = inject(ApiService);
  private readonly base = '/admin/system-health';

  subsystems() {
    return this.api.get<Subsystem[]>(`${this.base}/subsystems`);
  }

  kpis() {
    return this.api.get<SystemHealthKpis>(`${this.base}/kpis`);
  }

  events() {
    return this.api.get<SystemHealthEvent[]>(`${this.base}/events`);
  }

  getSubsystem(id: string) {
    return this.api.get<Subsystem>(`${this.base}/subsystems/${id}`);
  }

  probe(id: string) {
    return this.api.post<Subsystem>(`${this.base}/subsystems/${id}/probe`, {});
  }

  flushCache() {
    return this.api.post<void>(`${this.base}/cache/flush`, {});
  }

  getMaintenance() {
    return this.api.get<MaintenanceMode>(`${this.base}/maintenance`);
  }

  setMaintenance(enabled: boolean, message: string | null) {
    return this.api.put<MaintenanceMode>(`${this.base}/maintenance`, { enabled, message });
  }
}

File: backend/src/main/java/com/postwerk/service/AdminSystemHealthService.java
package com.postwerk.service;

import com.postwerk.dto.admin.MaintenanceModeResponse;
import com.postwerk.dto.admin.SubsystemHealthResponse;
import com.postwerk.dto.admin.SystemHealthEventResponse;
import com.postwerk.dto.admin.SystemHealthKpisResponse;

import java.util.List;
import java.util.UUID;

/**
 * Platform-staff System Health: live probes of Postwerk's own subsystems (API, PostgreSQL, Redis,
 * job scheduler, email-sync workers, SMTP, AI provider), plus gated re-probe / cache-flush /
 * maintenance-mode actions. Reads need {@code INFRA_VIEW}; mutations need {@code INFRA_MANAGE}.
 *
 * @since 1.0
 */
public interface AdminSystemHealthService {

    List<SubsystemHealthResponse> subsystems();

    SystemHealthKpisResponse kpis();

    List<SystemHealthEventResponse> events();

    /** Re-probes one subsystem by id; throws if the id is unknown. */
    SubsystemHealthResponse getSubsystem(String id);

    SubsystemHealthResponse probe(String id, UUID actorUserId, String ip);

    /** Clears the application cache regions (NOT the whole Redis instance). */
    void flushCache(UUID actorUserId, String ip);

    MaintenanceModeResponse getMaintenance();

    MaintenanceModeResponse setMaintenance(boolean enabled, String message, UUID actorUserId, String ip);
}
